lib/dock.rb

```diff
@@ -0,0 +1,20 @@
+class Dock
+  attr_reader :name, :max_rental_time, :revenue, :rentals
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @revenue = 0
+    @rentals = {}
+  end
+
+  def rent(boat, renter)
+    @rentals[boat] = renter
+  end
+
+  # def log_hour
+  #   @rentals.each do |rental|
```

Comment:
This looks really close! Since @rentals is a hash, you need to include two block variables for the key and value